### DISCUSSION TITLE
Is the database `macaron.db` incremental and cacheable?

### DISCUSSION THREAD

**Discussion by jgsuess** [oracle/macaron#348] [Q&A]
Given that macaron operations break into the distinct phases of collection and policy analysis, and that the artifact of importance is the `macaron.db` I am wondering if it is incremental and cacheable?

I.e. if I run collection (`analyze`) multiple times, will each run just add to an existing `macaron.db`? Can I hence cache the DB and run analysis only if I have new questions of the analysis?

**Comment by behnazh-w** [comment]
Yes, definitely. You should cache the database, i.e., `macaron.db`, which allows you to run the policy engine with small performance overhead subsequently.
